gc/gc-iceberg/src/main/java/org/apache/iceberg/ManifestReaderUtil.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.iceberg;
+
+import com.google.common.collect.ImmutableList;
+import org.apache.iceberg.io.CloseableIterable;
+import org.apache.iceberg.io.FileIO;
+
+public final class ManifestReaderUtil {
+
+  private ManifestReaderUtil() {}
+
+  public static CloseableIterable<String> readPathsFromManifest(ManifestFile manifest, FileIO io) {
+    // entry.file() is package private. Hence, this Util class under iceberg package.
+    return CloseableIterable.transform(
+        ManifestFiles.open(manifest, io, null).select(ImmutableList.of("file_path")).liveEntries(),
```

Comment:
Hard to convince the community. I will give it a try. 

Since we don't want to wait till the new Iceberg release for this issue to be fixed, we can still go ahead with this fix.

Reply:
`ManifestFiles.open()` is small and straight-forward. Why not inline it into Nessie code while we wait for the Iceberg PR?

Reply:
`Manifest.open()` is not really the problem. We can call its internal functions directly and no need of making it public.

`entry.file()` is the real problem

Reply:
I think a case could be made that `ManifestFiles.readPaths` should "just" work for delete files as well. I don't see a reason why it shouldn't (both data and delete manifests list paths, etc.). That would in some sense be a minimally invasive change on Iceberg's end and might be easier to convince the community on?

Reply:
After reviewing the callers of `ManifestFiles.readPaths`, it looks like Iceberg uses it for only expire snapshots with branches (ReachableFileCleanup). And it doesn't handle delete files :) 

So, we need to support `ManifestFiles.readPaths` properly and also handle delete files for that scenario. I will raise a separate PR at Iceberg side. But Iceberg release may take months. We can go ahead with current changes at Nessie side as a temporary solution.

Reply:
> We can probably call entry.file() via java reflection (since the method is public, even though the interface is not).
Not just `entry.file()` is package private. 
`ManifestReader#liveEntries()` which is used in `readPathsFromManifest` is also a package private method. The result of that method `ManifestEntry` is also a package private.  So, I cannot cast in our package after reflection. 

As mentioned above, I will work on long term solution at Iceberg repo. As a short term solution, we might have to go with iceberg package names in Nessie.

Reply:
If @snazy is ok with this, I'm ok :)